woocommerce/google-listings-and-ads
# Add Program Analytics page

### Changes proposed in this Pull Request:



Closes #96, #99, #100.

- Creates Program reports UI as described in https://github.com/woocommerce/google-listings-and-ads/issues/96
- Adds `wcadmin_gla_table_sort` track event for AppTableCard. ⚠️ It adds it also for Dashboard, Product Feed tables as well.
- Adds `wcadmin_gla_datepicker_update` track event for the DatePickerFilter. It adds it also for Dashboard


### Screenshots:

![Animation that shows tracking events appear in the console.](https://user-images.githubusercontent.com/17435/104525677-f4ae8980-5600-11eb-90fa-1b7691c58516.gif)
![Animation that shows UI changes for different Programs sets.](https://user-images.githubusercontent.com/17435/104525706-0c860d80-5601-11eb-9e86-7ff868489441.gif)
![Animation that shows UI changes for different screen sizes.](https://user-images.githubusercontent.com/17435/104525899-7e5e5700-5601-11eb-9fc8-26b940b68113.gif)




### Detailed test instructions:

1. Go to http://localhost/wordpress/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports
2. Check if the UI matches the [Figma designs](https://www.figma.com/file/jZUpa8eTrnrK1Lwt2ry7zk/Native-Google-Integration?node-id=2220%3A1324) and requirements in https://github.com/woocommerce/google-listings-and-ads/issues/96
   - Filters have required options,
   - Metrics are presented correctly
     You can use via temporary, debugging URL param `missingFreeListingsData` to switch mocked data, to test 3 scenarios:
 
      - not set or `&missingFreeListingsData=` - free and paid listings with no missing data (2022 case)
	  - `&missingFreeListingsData=na` only free listings without some metrics (pre Q4 2021, no paid programs)
	  - `&missingFreeListingsData=anythingelse` free and paid programs with missing data (pre Q4 2021, with paid programs)
   - Compare Programs table looks correctly,
   - Track events are being fired (You can execute `localStorage.setItem( 'debug', 'wc-admin*' );` in the browser console to get them logged)
      - `wcadmin_gla_table_header_toggle`
      - `wcadmin_gla_table_sort`
	  - `wcadmin_gla_datepicker_update`


### Changelog Note:

> (Optional) Enter a summary of all changes in this Pull Request to appear in the changelog if it's accepted. 


### Follow-ups:
- Performance metric's deltas / `<SummaryNumber>` does not render numbers with given precision https://github.com/woocommerce/woocommerce-admin/issues/5925
- `<CheckboxControl>` does collapse its right margin, making it impossible to center it, without knowing the implementation details - issue TBC
- Single and Compare program filters, need to be implemented - follow-up issue TBC, related ones https://github.com/woocommerce/woocommerce-admin/issues/6061, https://github.com/woocommerce/woocommerce-admin/issues/6062
    - Plus track event `gla_filter`
- Track event for switching chart mode `gla_chart_tab_click` - `<Chart>` does not expose a nice way to observe when the char type is switched.
- There is some work pending on the API side. These are marked with a "TODO" comment in the code.

### Aditional comments:
 - [One line](https://github.com/woocommerce/google-listings-and-ads/pull/109/files#diff-382a32c266ad3f58132a6e23bd8c2655697bad60fbd0dccabf8fe3c322b80236R23) still makes eslint unhappy. `@return {module:@woocommerce/components~DateRangeFilterPicker}` 
    I guess it's 
    - old version of JSDoc parser
    - @woocommerce/components not providing docs and type.
    
    We can either leave it as is and wait for the times we will improve our (JS)Docs, remove that line, or create a follow-up.
